bgls/near_clifford_solver_test.py

```diff
@@ -0,0 +1,98 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the BGLS Simulator near clifford solver."""
+
+import cirq
+
+import bgls
+import bgls.near_clifford_solver
+
+
+def test_expanded_clifford():
+    """Each circuit in circuit_clifford_decomposition should be pure
+    clifford"""
+    qubits = cirq.LineQubit.range(3)
+    domain = {cirq.H, cirq.CNOT, cirq.S, cirq.T}
+    clifford_circuit = bgls.utils.generate_random_circuit(
+        qubits,
+        n_moments=10,
+        op_density=0.5,
+        gate_domain=domain,
+        random_state=1,
+    )
+    (
+        expanded_circuits,
+        expanded_amplitudes,
+    ) = bgls.near_clifford_solver.circuit_clifford_decomposition(
+        clifford_circuit
+    )
+
+    for circuit in expanded_circuits:
+        for op in circuit.all_operations():
+            assert cirq.has_stabilizer_effect(op)
```

Comment:
I think you can call `cirq.has_stabilizer_effect` on the circuit? Might be a little faster, and cleaner to read.

Reply:
I tried this and it doesn't work - I looked into it and it tries applying different application strats so it works at the gate level only. If we find a cleaner way though I will definitely use